opentelemetry-api/src/opentelemetry/metrics/__init__.py

```diff
@@ -11,3 +11,513 @@
+    aggregated data. This should be used to report metrics like cpu/memory
+    usage, in which the type of aggregation is already defined, or simple
+    metrics like "queue_length".
+
+    Initialization of the :class:`Meter` is done through the `meter()` function
+    (see below). The meter instance has a singleton implementation.
+    """
+
+    def create_double_counter(self,
+                              name: str,
+                              options: typing.Optional['MetricOptions'] = None
+                              ) -> 'CounterDouble':
+        """Creates a counter type metric that contains double values.
+
+        Args:
+            name: The name of the counter.
+            options: An optional :class:`.MetricOptions` used to populate
+            details about the counter.
+
+        Returns: A new :class:`.CounterDouble`
+        """
+
+    def create_long_counter(self,
+                            name: str,
+                            options: typing.Optional['MetricOptions'] = None
+                            ) -> 'CounterLong':
+        """Creates a counter type metric that contains long values.
+
+        Args:
+            name: The name of the counter.
+            options: An optional :class:`.MetricOptions` used to populate
+            details about the counter.
+
+        Returns:
+            A new :class:`.CounterLong`
+        """
+
+    def create_double_gauge(self,
+                            name: str,
+                            options: typing.Optional['MetricOptions'] = None
+                            ) -> 'GaugeDouble':
+        """Creates a gauge type metric that contains double values.
+
+        Args:
+            name: The name of the gauge.
+            options: An optional :class:`.MetricOptions` used to populate
+            details about the gauge.
+
+        Returns:
+            A new :class:`.GaugeDouble`
+        """
+
+    def create_long_gauge(self,
+                          name: str,
+                          options: typing.Optional['MetricOptions'] = None
+                          ) -> 'GaugeLong':
+        """Creates a gauge type metric that contains long values.
+
+        Args:
+            name: The name of the gauge.
+            options: An optional :class:`.MetricOptions` used to populate
+            details about the gauge.
+
+        Returns:
+            A new :class:`.GaugeLong`
+        """
+
+    def create_measure(self,
+                       name: str,
+                       options: typing.Optional['MeasureOptions'] = None
+                       ) -> 'Measure':
+        """Creates a Measure used to record raw :class:`.Measurement`s.
+
+        Args:
+            name: the name of the measure
+            options: An optional :class:`.MeasureOptions` used to populate
+            details about the measure
+
+        Returns:
+            A :class:`.Measure`
+        """
+
+    def record(self,
```

Comment:
Might be worth discussing whether we need constant labels for measurements since `distributed_context` is not explicit. I'll bring it up in the meeting.